HDF5Examples/C/H5FLT/h5ex_d_bitgroom.c

```diff
@@ -0,0 +1,239 @@
+    float        wdata[DIM0][DIM1], /* Write buffer */
+        rdata[DIM0][DIM1],          /* Read buffer */
+        max;
+    hsize_t i, j;
+    int     ret_value = 1;
+
+    /*
+     * Initialize data.
+     */
+    for (i = 0; i < DIM0; i++)
+        for (j = 0; j < DIM1; j++)
+            wdata[i][j] = i * j - j;
+
+    /*
+     * Create a new file using the default properties.
+     */
+    file_id = H5Fcreate(FILE, H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT);
+    if (file_id < 0)
+        goto done;
+
+    /*
+     * Create dataspace.  Setting maximum size to NULL sets the maximum
+     * size to be the current size.
+     */
+    space_id = H5Screate_simple(2, dims, NULL);
+    if (space_id < 0)
+        goto done;
+
+    /*
+     * Create the dataset creation property list, add the BitGroom
+     * quantization filter and set the chunk size.
+     */
+    dcpl_id = H5Pcreate(H5P_DATASET_CREATE);
+    if (dcpl_id < 0)
+        goto done;
+
+    /* 20200929: csz change this to H5Z_FLAG_OPTIONAL, so that can_apply() can reject filter for
+       integers without causing program to exit()? */
+    status = H5Pset_filter(dcpl_id, H5Z_FILTER_BITGROOM, H5Z_FLAG_MANDATORY, nelmts, cd_values);
+    if (status < 0)
+        goto done;
+
+    /*
+     * Check that filter is registered with the library now.
+     * If it is registered, retrieve filter's configuration.
+     */
+    avail = H5Zfilter_avail(H5Z_FILTER_BITGROOM);
+    if (avail) {
+        status = H5Zget_filter_info(H5Z_FILTER_BITGROOM, &filter_config);
+        if ((filter_config & H5Z_FILTER_CONFIG_ENCODE_ENABLED) &&
+            (filter_config & H5Z_FILTER_CONFIG_DECODE_ENABLED))
+            printf("BitGroom filter is available for quantization and decoding.\n");
+    }
+    else {
+        printf("H5Zfilter_avail - not found.\n");
+        goto done;
+    }
+    status = H5Pset_chunk(dcpl_id, 2, chunk);
+    if (status < 0)
+        printf("failed to set chunk.\n");
+
+    /*
+     * Create the dataset.
+     */
+    printf("....Create dataset ................\n");
+    dset_id = H5Dcreate(file_id, DATASET, H5T_IEEE_F32LE, space_id, H5P_DEFAULT, dcpl_id, H5P_DEFAULT);
+    if (dset_id < 0) {
+        printf("failed to create dataset.\n");
+        goto done;
+    }
+
+    /*
+     * Write the data to the dataset.
+     */
+    printf("....Writing BitGroom-quantized data ................\n");
+    //    status = H5Dwrite (dset_id, H5T_NATIVE_FLOAT, H5S_ALL, H5S_ALL, H5P_DEFAULT, wdata[0]);
+    status = H5Dwrite(dset_id, H5T_NATIVE_FLOAT, H5S_ALL, H5S_ALL, H5P_DEFAULT, (void *)wdata);
+    if (status < 0)
+        printf("failed to write data.\n");
+
+    /*
+     * Close and release resources.
+     */
+    H5Dclose(dset_id);
+    dset_id = -1;
+    H5Pclose(dcpl_id);
+    dcpl_id = -1;
+    H5Sclose(space_id);
+    space_id = -1;
+    H5Fclose(file_id);
+    file_id = -1;
+    status  = H5close();
+    if (status < 0) {
+        printf("/nFAILED to close library/n");
```

Comment:
`/` -> `\` (May want to check if this is a copy-paste from other examples)

Reply:
This is from the original source

Reply:
Sure but it's not a bad idea to go ahead and fix it since it's wrong

Reply:
It then needs to fixed in the plugins code first

Reply:
But then all the examples in the plugins will need to fixed because they all do the same

Reply:
Which is why:

> May want to check if this is a copy-paste from other examples

Not exactly a big deal but definitely a why not

Reply:
Yes it is probably all cut-n-paste from the beginning

Reply:
Might as well fix them all at once separately I guess